go/kbfs/libkbfs/block_disk_store.go

```diff
@@ -67,11 +69,20 @@ import (
 //
 //   /01ff/f...(30 characters total)...ff/data
 //
-// blockDiskStore is not goroutine-safe, so any code that uses it must
-// guarantee that only one goroutine at a time calls its functions.
+// blockDiskStore is goroutine-safe on a per-operation, per-block ID
+// basis, so any code that uses it can make concurrent calls.
+// However, it's likely that much of the time, the caller will require
+// consistency across multiple calls (i.e., one call to
+// `getDataSize()`, followed by a call to `remove()`), so higher-level
+// locking is also recommended.  For performance reasons though, it's
+// recommended that the `put()` method can be called concurrently, as
+// needed.
```

Comment:
@jzila I'm not sure I follow.  What's the argument for needing a more specific name here?  Is it ambiguous with something else?  Maybe you're thinking of the disk cache? (I consider store vs. cache to be clear enough, but maybe not everyone does...)

Reply:
I'm just trying to capture what the lock is guarding, since it's kind of a weird one where you get a read lock in order to put, and a write lock in order to flush.

Reply:
Oh, you weren't talking about renaming the whole struct?  Sure, I'm happy to rename the lock.

Reply:
Oh wait I asked for both, I thought I was responding to the lock rename. In this case, I was suggesting a rename because we'll be moving to a new disk store approach for the journal, so we'll have more than one of these soon.

Reply:
Ah got it.  I'd rather do that rename, if needed, in the PR that introduces the new storage system.  I'm not even sure this struct would survive in that PR.  Perhaps its innards would just be replaced with leveldb stuff.

Reply:
Sounds good to me!